pyshelf/search/manager.py

```diff
@@ -30,109 +30,52 @@ def search(self, criteria, key_list=None):
+                            SortFlag.VERSION
                         ]
                     }
                 ]
             }
         """
-        query = Search().index(Metadata._doc_type.index).query(self._build_query(criteria["search"]))
-        results = query.execute()
-        filtered_results = self._filter_results(results.hits, key_list)
-        formated_results = self._sort_results(filtered_results, criteria.get("sort"))
+        query = self._build_query(criteria.get("search"))
+        query = Search(using=self.connection).index(self.search_container.es_index).query(query)
+        self.search_container.logger.debug("Executing the following search query: {0}".format(query.to_dict()))
+        search_results = query.execute()
+        search_formatter = SearchFormatter(criteria, search_results, key_list)
 
-        return formated_results
-
-    def _sort_results(self, filtered_results, sort_criteria=None):
-        """
-            Sorts results based on sort criteria.
-
-            Args:
-                filtered_results(list(dict)): list of dictionaries representing hits from Elasticsearch.
-
-            Returns:
-                list(dict): Sorts the filtered results that are passed in.
-        """
-        if not sort_criteria:
-            return filtered_results
-
-        for criteria in sort_criteria:
-            kwargs = {}
-            if SortType.DESC in criteria["flag_list"]:
-                kwargs["reverse"] = True
-
-            if criteria.get("sort_type") == SortType.VERSION:
-                filtered_results.sort(key=lambda k: LooseVersion(k[criteria["field"]]["value"]), **kwargs)
-            else:
-                filtered_results.sort(key=lambda k: k[criteria["field"]]["value"], **kwargs)
-
-        return filtered_results
-
-    def _filter_results(self, hits, key_list=None):
-        """
-            Filters results into a consumable format.
-
-            Args:
-                hits(list(elasticsearch_dsl.Result): List of result objects as returned from Response.hits
-                key_list(list): List of keys to return. None assumes all keys are requeired
-
-            Returns:
-                list(dict): Formats result list to a list of dictionaries. Each element of the list representing a hit.
-        """
-        wrapper = []
-
-        for hit in hits:
-            filtered = {}
-            add = True
-
-            for item in hit.items:
-                if item.name in self.tilde_search.keys() and LooseVersion(item.value) < \
-                        LooseVersion(self.tilde_search[item.name]):
-                    add = False
-
-                if key_list:
-
-                    if item["name"] in key_list:
-                        filtered.update({item["name"]: item})
-                else:
-                    filtered.update({item["name"]: item})
-
-            if add:
-                wrapper.append(filtered)
-
-        return wrapper
+        return search_formatter.results
```

Comment:
For sure. Not sure what I was smoking here. a90d5928b76b46d09034bee53733a1c3df043a25